src/panel/direction/DirectionPanel.jsx

```diff
@@ -56,11 +59,24 @@ export default class DirectionPanel extends React.Component {
     this.restorePoints(props);
   }
 
-  componentDidMount() {
+  async componentDidMount() {
     Telemetry.add(Telemetry.ITINERARY_OPEN);
     document.body.classList.add('directions-open');
     this.dragPointHandler = listen('change_direction_point', this.changeDirectionPoint);
     this.setPointHandler = listen('set_direction_point', this.setDirectionPoint);
+
+    if (!this.props.origin && isMobileDevice()) {
+      // If authorized, set origin to current position on mobile
+      const isGeolocAvailable = await new Promise(resolve =>
```

Comment:
It seems this is not blocking, as the field gets filled with "My position" before the user has actually accepted to enable it. As a result, even if here refuses it will still be there.

Reply:
Yeah I'm on it